src/main/java/org/jabref/gui/externalfiles/FindFullTextAction.java

```diff
@@ -71,13 +74,22 @@ public void execute() {
         }
 
         Task<Map<BibEntry, Optional<URL>>> findFullTextsTask = new Task<Map<BibEntry, Optional<URL>>>() {
+
             @Override
             protected Map<BibEntry, Optional<URL>> call() {
                 Map<BibEntry, Optional<URL>> downloads = new ConcurrentHashMap<>();
                 int count = 0;
-                for (BibEntry entry : basePanel.getSelectedEntries()) {
-                    FulltextFetchers fetchers = new FulltextFetchers(Globals.prefs.getImportFormatPreferences());
-                    downloads.put(entry, fetchers.findFullTextPDF(entry));
+                AutoSetFileLinksUtil util = new AutoSetFileLinksUtil(basePanel.getBibDatabaseContext(), Globals.prefs.getFilePreferences(), Globals.prefs.getAutoLinkPreferences(), ExternalFileTypes.getInstance());
+                List<BibEntry> changed = util.linkAssociatedFiles(basePanel.getSelectedEntries(), new NamedCompound(""));
+
+                for (BibEntry entry : changed) {
```

Comment:
why do you only run over the changed entries?

Reply:
It's a bit misleading, the method returns all fields it gets as input and just adds the entry file list.
So these are exactly the same entries as from getSelectedEntries just with addtinal file fields if anything found,